types/time.go

```diff
@@ -493,9 +471,9 @@ func (t *Time) Check() error {
 func (t *Time) Sub(t1 *Time) Duration {
 	var duration gotime.Duration
 	if t.Type == mysql.TypeTimestamp && t1.Type == mysql.TypeTimestamp {
-		a, err := t.Time.GoTime(t.TimeZone)
+		a, err := t.Time.GoTime(gotime.UTC)
```

Comment:
It's `gotime.UTC` at the very beginning, but there is a bug related here, so it's changed to `t.TimeZone`...

Reply:
See https://github.com/pingcap/tidb/pull/5424
I'm not sure 1970-04-01 sub 1970-01-01 in America is the same as which in China.
So I wonder if we can pass an extra statement context to Sub and use the TimeZone in it?